### PR TITLE
Move dagrun_operator.py to trigger_dagrun.py (#11178)

### PR DESCRIPTION
**Pull request by kishvanchee**
This change is related to #11178 

**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by kishvanchee** [comment]
@mik-laj Please review and let me know if I have to change anything.

File: airflow/operators/dagrun_operator.py [hunk 1]
```diff
@@ -15,88 +15,14 @@
...
-from airflow.api.common.experimental.trigger_dag import trigger_dag
-from airflow.models import BaseOperator, BaseOperatorLink, DagRun
-from airflow.utils import timezone
-from airflow.utils.decorators import apply_defaults
-from airflow.utils.types import DagRunType
+# pylint: disable=unused-import
+from airflow.operators.dagrun import TriggerDagRunOperator # noqa
 
-
-class TriggerDagRunLink(BaseOperatorLink):
```
Comment (turbaszek) [inline airflow/operators/dagrun_operator.py:33]: You missed moving this class as well as operator_extra_links in the `TriggerDagRunOperator`

**Review by turbaszek — CHANGES_REQUESTED** [review]

File: airflow/operators/dagrun_operator.py (hunk 1 above)
Comment (kishvanchee) [reply airflow/operators/dagrun_operator.py:33]: Got it. Just noticed. I guess I didn't read the conflict properly although when I was working on it the particular `class TriggerDagRunLink(BaseOperatorLink)` didn't exist in the file. I'll fix this now.

Please correct me if I'm wrong, so to fix this, I should finish working on the file, fetch/rebase, then push again. Right?

Comment (kishvanchee) [reply airflow/operators/dagrun_operator.py:33]: I have pushed with the corrections. @turbaszek please have a look.

Apologies for the previous commit. I messed up with the rebase.

**Review by turbaszek — COMMENTED** [review]
Look good but one more change here will be needed:
https://github.com/apache/airflow/blob/baa980fbc83b25b5cf0700e567e69c2eb156412f/airflow/serialization/serialized_objects.py#L57

**Comment by kishvanchee** [comment]
I just noticed. In tests it is named as `test_trigger_dagrun_operator_conf` and `test_trigger_dagrun_operator_templated_conf`. Should those be changed to `test_trigger_dagrun_conf` and `test_trigger_dagrun_templated_conf`

**Comment by mik-laj** [comment]
> I just noticed. In tests it is named as test_trigger_dagrun_operator_conf and test_trigger_dagrun_operator_templated_conf. Should those be changed to test_trigger_dagrun_conf and test_trigger_dagrun_templated_conf ?

This is not necessary in my opinion. 

> Move dagrun_operator.py to dagrun.py  

I have doubts if this is a good name for a module. Wouldn't `trigger_dagrun` be better? @turbaszek WDYT

**Comment by kishvanchee** [comment]
I think I should be updated `operators-and-hooks-ref.rst` too. I'll await further comments for now before doing that.

**Comment by turbaszek** [comment]
> Wouldn't trigger_dagrun be better?

+1 for this name  and I second Kamil that there's no need to change names of tests but yeah, we need update in `operators-and-hooks-ref.rst` and everywhere in documentation (I would use IDE to look for this operator in the whole Airflow code base)

**Comment by turbaszek** [comment]
> I started `find`ing `.rst` only recently now. 

And we need to update those files too. We should use new path not only in code but also in documentation.

I'm personally using PyCharm which has a refactor option that automatically searches for related occurrences and fixes them. This helps a lot with automating. Eventually one can use `cmd + r` to find and replace a fragment of text - this would do the trick in this case as we are only changing import path

**Comment by kaxil** [comment]
Can you please rebase your PR on latest Master since we have applied [Black](https://github.com/apache/airflow/commit/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68) and [PyUpgrade](https://github.com/apache/airflow/commit/8c42cf1b00c90f0d7f11b8a3a455381de8e003c5) on Master.

It will help if your squash your commits into single commit first so that there are less conflicts.

**Comment by turbaszek** [comment]
@kishvanchee any update on this one?

**Comment by kishvanchee** [comment]
@turbaszek @kaxil Hi, I apologize for the delay. I'll update this by tomorrow afternoon IST.

**Comment by kishvanchee** [comment]
@kaxil @turbaszek Can you please have a look? I have rebased and merged.

**Review by kaxil — APPROVED** [review]

**Review by turbaszek — APPROVED** [review]

**Comment by ashb** [comment]
Closing in favour of https://github.com/apache/airflow/pull/12933

**Pull request closed without merging**